### ISSUE TITLE
Docs: Improve examples on overriding deployment resources

### ISSUE THREAD

**Issue by oraNod** [ansible-community/awx-operator-helm#29]
Follows up on https://github.com/ansible-community/awx-operator-helm/pull/20#discussion_r1854039069

Purpose is to improve documentation in README at https://github.com/ansible-community/awx-operator-helm/blob/70bac70f140c05719f21a4faa292c2805d1fb152/README.md?plain=1#L249

While the example of the replica is valid. This is not really a common scenario to customize the operator replica (maybe to scale up/scale down). One more common use case is to specify a different container image for each container as seen in https://github.com/ansible-community/awx-operator-helm/issues/17 or https://github.com/ansible-community/awx-operator-helm/issues/27 to cover disconnected/restricted environments.
Another useful example would be to provide an example for setting up resource requests and limits, this would cover https://github.com/ansible-community/awx-operator-helm/issues/14.

**Comment by miles-w-3** [comment]
Adding additional context from the other thread as well, This is more of a feature because the naive override approach would have to be overriding the entire container list since helm does not support merging lists together.

The alternative is to improve the templating logic to offer a `containerConfigOverride` key or something similar where you can provide a key which matches a container name and the values you want to layer in, and then helm template logic to loop through the containers list from the original operator and splice the override configs in. This is something I can work on once we get releases going


https://github.com/ansible-community/awx-operator-helm/pull/20#discussion_r1882476669

**Comment by miles-w-3** [comment]
Will be closed in #48

**Comment by wannte** [comment]
Hi, @miles-w-3. 

Once this task is completed, will it be easy to modify the CPU/memory request and limit for the `kube-rbac-proxy` and `awx-manager` within the `awx-operator-controller-manager` pod? 
If so, thank you. This is the feature I have been wanting and looking for.

**Comment by miles-w-3** [comment]
yes, it will allow you to override any spec options of the operator-controller deployment as needed
